From 598831d1132718e853724c0013513a3b7ead620d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 14:58:57 +0800
Subject: [PATCH 261/300] AOSCOS: staging: sb105x: replace deprecated strlcpy()
 with strscpy()

The string function strlcpy() was removed in commit d26270061ae6 ("string:
Remove strlcpy()") and per deprecated.rst:

  strlcpy() reads the entire source buffer first (since the return value
  is meant to match that of strlen()). This read may exceed the
  destination size limit. This is both inefficient and can lead to linear
  read overflows if a source string is not NUL-terminated. The safe
  replacement is strscpy(), though care must be given to any cases where
  the return value of strlcpy() is used, since strscpy() will return
  negative errno values when it truncates.

Replace calls to strlcpy() with strscpy().

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index 963f3091a5c9..cd98d3b0183d 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -1624,7 +1624,7 @@ static inline void mp_report_port(struct uart_driver *drv, struct sb_uart_port *
 			break;
 		default:
 			snprintf(address, sizeof(address),"*unknown*" );
-			strlcpy(address, "*unknown*", sizeof(address));
+			strscpy(address, "*unknown*", sizeof(address));
 			break;
 	}
 
-- 
2.48.1

